lib/internal/modules/esm/resolve.js

```diff
@@ -374,18 +374,21 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
       resolved.pathname, 'must not include encoded "/" or "\\" characters',
       fileURLToPath(base));
 
-  const path = fileURLToPath(resolved);
+  let path = fileURLToPath(resolved);
   if (getOptionValue('--experimental-specifier-resolution') === 'node') {
     let file = resolveExtensionsWithTryExactName(resolved);
-    if (file !== undefined) return file;
-    if (!StringPrototypeEndsWith(path, '/')) {
-      file = resolveDirectoryEntry(new URL(`${resolved}/`));
-      if (file !== undefined) return file;
-    } else {
-      return resolveDirectoryEntry(resolved) || resolved;
+
+    if (file === undefined) {
+      file = StringPrototypeEndsWith(path, '/') ?
+        resolveDirectoryEntry(resolved) : resolveDirectoryEntry(new URL(`${resolved}/`));
     }
-    throw new ERR_MODULE_NOT_FOUND(
-      resolved.pathname, fileURLToPath(base), 'module');
+
+    if (file === undefined) {
```

Comment:
This is "if it's still undefined after attempting to resolve it"

Reply:
What I meant is instead of
```js
if (file === undefined) {
 // do something
}
if (file === undefined) {
 // if it's still undefined
}
```
we could have:
```js
if (file === undefined) {
  // do something
  if (file === undefined) {
    // if it's still undefined
  }
}
```